src/eager.js

```diff
@@ -6,8 +6,14 @@ import Helpers from './helpers';
 import Promise from './base/promise';
 import EagerBase from './base/eager';
 
-const getAttributeUnique = (models, attribute) =>
-  _.uniq(_.map(models, m => m.get(attribute)));
+const getAttributeUnique = (models, attribute) => {
+  return _.uniq(_.map(models, m => {
+    // use formatted attributes so that morphKey and foreignKey will match
+    // attribute keys
+    const attributes = m.format(_.clone(m.attributes));
```

Comment:
@vellotis I can't `pick` the right keys because they don't match the parsed attributes. It's why I need to `format` the attributes first. As I said in the description I could `parse` the morphKeys instead of `format` the attributes but I'm not familiar enough with the codebase so I did the same  than in [relation.js](https://github.com/tgriesser/bookshelf/blob/master/src/relation.js#L29). If you prefer I can do something like that : 

```js
const parsedAttribute = _.head(m.parse({ attribute }));
return m.get(parsedAttribute);
```